crates/core/src/operations/transaction/mod.rs

```diff
@@ -293,19 +302,27 @@ impl CommitData {
     }
 }
 
+#[derive(Clone, Debug, Copy)]
+/// Properties for post commit hook.
+pub struct PostCommitHookProperties {
+    create_checkpoint: bool,
+}
+
```

Comment:
I think we can remove this struct and just keep the config in the commit builder and commit properties for now.

Reply:
I was planning to add support for log retention cleanup, auto compact and auto vacuüm afterwards. So perhaps still good to keep this then?

Reply:
Sound good in that case.